sky/optimizer.py

```diff
@@ -303,7 +304,13 @@ def _estimate_nodes_cost_or_time(
 
                     if minimize_cost:
                         cost_per_node = resources.get_cost(estimated_runtime)
-                        estimated_cost_or_time = cost_per_node * node.num_nodes
+                        available_reservation_resources = (
+                            resources.get_available_reservation_resources())
+                        # We consider the cost of the unused reservation
+                        # resources to be 0 since we are already paying for
+                        # them.
```

Comment:
Maybe we could add a TODO here, like the following
```suggestion
                        # We consider the cost of the unused reservation
                        # resources to be 0 since we are already paying for
                        # them.
                        # TODO: different policies can be applied here for
                        # whether to choose reserved instances.
```